### ISSUE TITLE
During inference, the output of noisy gate is nan.

### ISSUE THREAD

**Issue by zqhang** [laekov/fastmoe#180]
The training process proceeds smoothly; however, an issue arises during inference as the **noise_stddev** becomes zero when **self.training** is False, leading to an error when computing the **load**. Should we refrain from adding noise in the NoisyGate during inference?

**Comment by zqhang** [comment]
Do I accurately comprehend your statement:  `noise_stddev = self.softplus(raw_noise_stddev) * self.training + self.noise_epsilon` ?

**Comment by laekov** [comment]
> Do I accurately comprehend your statement: `noise_stddev = self.softplus(raw_noise_stddev) * self.training + self.noise_epsilon` ?

Yes, I think that can help fixing your nan issue. But as I am not an algiorithm person, I am not sure if this is what the nosiy gate is expected to behave for inference.

**Comment by zqhang** [comment]
Thank you for your help
